engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -76,6 +76,33 @@
   private String recordingNameTopic;
   private long numWindows = Long.MAX_VALUE; // number of windows to record
   private Runnable stopProcedure; // stop procedure to execute
+
+  private static final Logger logger = LoggerFactory.getLogger(TupleRecorder.class);
+
+  // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
+  // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
+  // log every tuple error
+  private static int ERROR_LOG_GAP;
+  int lastLog = -1;
+
+  static {
```

Comment:
The cost of getting property and converting it to int is negligible, I don't see why it is so important to initialize the gap only once, especially that logically every TupleRecorder may have different gaps due to a different number of tuples it records. Even though this functionality is not supported now (as we are using system properties), it may be supported in the future.